lib/translate/index.js

```diff
@@ -0,0 +1,302 @@
+ *     //     input: 'Hola'
+ *     //   }
+ *     // ]
+ *   }
+ * });
+ */
+Translate.prototype.detect = function(input, callback) {
+  input = arrify(input);
+
+  this.request({
+    uri: '/detect',
+    useQuerystring: true,
+    qs: {
+      q: input
+    }
+  }, function(err, resp) {
+    if (err) {
+      callback(err, null, resp);
+      return;
+    }
+
+    var results = resp.data.detections.map(function(detection, index) {
+      var result = extend(detection[0], {
+        input: input[index]
+      });
+
+      // Deprecated.
+      delete result.isReliable;
+
+      return result;
+    });
+
+    if (input.length === 1) {
+      results = results[0];
+    }
+
+    callback(null, results, resp);
+  });
+};
+
+/**
+ * Get an array of all supported languages.
+ *
+ * @resource [Discover Supported Languages]{@link https://cloud.google.com/translate/v2/using_rest#supported-languages}
+ *
+ * @param {function} callback - The callback function.
+ * @param {?error} callback.err - An error returned while making this request.
+ * @param {string[]} callback.languages - The supported ISO 639-1 language
+ *     codes.
+ * @param {object} callback.apiResponse - Raw API response.
+ *
+ * @example
+ * translate.getLanguages(function(err, languages) {
+ *   if (!err) {
+ *     // languages = [
+ *     //   'af',
+ *     //   'ar',
+ *     //   'az',
+ *     //   ...
+ *     // ]
+ *   }
+ * });
+ */
+Translate.prototype.getLanguages = function(callback) {
+  this.request({
+    uri: '/languages'
+  }, function(err, resp) {
+    if (err) {
+      callback(err, null, resp);
+      return;
+    }
+
+    var languages = resp.data.languages.map(prop('language'));
+    callback(null, languages, resp);
+  });
+};
+
+/**
+ * Translate a string or multiple strings into another language.
+ *
+ * @param {string|string[]} input - The source string input.
+ * @param {string} language - The target ISO 639-1 language code to translate
+ *     the source input to. (e.g. `en` for English)
+ * @param {function} callback - The callback function.
+ * @param {?error} callback.err - An error returned while making this request.
+ * @param {object|object[]} callback.translations - If a single string input was
+ *     given, a single translation is given. Otherwise, it is an array of
+ *     translations.
+ * @param {object} callback.apiResponse - Raw API response.
+ *
+ * @example
+ * translate.translate('Hello', 'es', function(err, translation) {
+ *   if (!err) {
+ *     // translation = 'Hola'
+ *   }
+ * });
+ *
+ * //-
+ * // Translate multiple strings of input. Note that the results are
+ * // now provied as an array.
+ * //-
+ * var input = [
+ *   'Hello',
+ *   'How are you today?'
+ * ];
+ *
+ * translate.translate(input, 'es', function(err, translations) {
+ *   if (!err) {
+ *     // translations = [
+ *     //   'Hola',
+ *     //   'Como estas hoy?'
+ *     // ]
+ *   }
+ * });
+ */
+Translate.prototype.translate = function(input, language, callback) {
```

Comment:
I didn't say anything about it to see if it bugged you too :) I'm struggling to think of something better. Super open to it though!